akka-docs/rst/scala/code/docs/http/scaladsl/server/directives/DebuggingDirectivesExamplesSpec.scala

```diff
@@ -95,4 +95,36 @@ class DebuggingDirectivesExamplesSpec extends RoutingSpec {
       responseAs[String] shouldEqual "logged"
     }
   }
+  "logRequestResultWithResponseTime" in {
+
+    def akkaResponseTimeLoggingFunction(loggingAdapter: LoggingAdapter,
+                                        requestTimestamp: Long,
+                                        level: LogLevel = Logging.InfoLevel)(req: HttpRequest)(res: Any): Unit = {
+      val entry = res match {
+        case Complete(resp) =>
+          val responseTimestamp: Long = System.currentTimeMillis()
```

Comment:
`currentTimeMillis` is actually not guaranteed to always increase (system adjusting the clock from NTP after drift for example), the safe elapsed-time tool would be `System.nanoTime()`, it will always only increase (it returns nanoseconds so you'd have to recalculate into millis or possibly seconds for the log entry).